wp-content/civi-extensions/goonjcustom/cli/assign-group-to-individual.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * @file
+ * CLI Script to assign contacts from a group to their respective state groups.
+ */
+
+use Civi\Api4\Contact;
+use Civi\Api4\Group;
+use Civi\Api4\GroupContact;
+
+if (php_sapi_name() != 'cli') {
+  exit("This script can only be run from the command line.\n");
+}
+
+// Change this ID to the one where you want to add the contact.
+define('SOURCE_GROUP_ID', 25);
+
+echo "Fetching contacts from group ID " . SOURCE_GROUP_ID . "...\n";
+
+/**
+ * Fetch contacts from the specified group.
+ */
+function getContactsFromGroup(): array {
+  $groupContacts = GroupContact::get(FALSE)
+    ->addSelect('contact_id')
+    ->addJoin('Contact AS contact', 'LEFT')
+    ->addWhere('group_id', '=', SOURCE_GROUP_ID)
+    ->execute();
+
+  return $groupContacts->getIterator()->getArrayCopy();
```

Comment:
Why getIterator?

Reply:
Use $groupContacts->getIterator()->getArrayCopy(); to convert the result into a plain PHP array. 
The ->execute(); method returns an object, not a regular array, so we use getIterator()->getArrayCopy() to extract the data in a standard format.